src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs

```diff
@@ -803,7 +804,16 @@ static TraceContextBase GetParentTraceContext(IList<MessageData> messages)
 
                             break;
                     }                   
-                }               
+                } else
+                {
+                    if (message.TaskMessage.Event is EventRaisedEvent)
+                    {
+                        var history = session.RuntimeState.Events;
+
+                        var traceContextString = history.Where(p => p.EventType == EventType.ExecutionStarted).Select(p => (ExecutionStartedEvent)p).FirstOrDefault().Correlation;
```

Comment:
Why do you need to search through this history? Can't you do this?
```suggestion
                        string traceContextString = session.RuntimeState.ExecutionStartedEvent?.Correlation;
```